src/modules/attempts/attempts.service.ts

```
@@ -112,6 +116,293 @@ export class AttemptsService {
+    const questionsWithoutSection = [];
+
+    for (const testQuestion of testQuestions) {
+      const question = questions.find(q => q.questionId === testQuestion.questionId);
+      if (!question) continue;
+
+             const userAnswer = answersMap.get(question.questionId);
+       
+       // Transform user answer to the desired format
+       let selectedOptions = null;
+       
+       if (userAnswer && userAnswer.answer) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)